File: Math/Sqrt(x)/sqrt.java
class Solution {
    public int mySqrt(int x) {
        if (x < 2) {
            return x;
        }
        long start = 1;
        long end = x;

        while (start <= end) {
            long mid = (start + end) / 2;

            if (mid * mid == x) {
                return (int) mid;
            } else if (mid * mid > x) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }
        return (int) end;
    }
}

File: Math/Sqrt(x)/sqrt.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x == 0 or x == 1:
            return x

        guess = x
        while abs(x - guess * guess) > 1e-6:
            guess = (guess + (x / guess)) / 2
        return math.floor(guess)
